hcx-apis/src/main/java/org/swasth/hcx/service/NotificationService.java

```diff
@@ -0,0 +1,132 @@
+
+    private List<Map<String, Object>> notificationList = null;
+
+    @PostConstruct
+    public void init() throws IOException {
+        loadNotifications();
+    }
+
+    public void processSubscription(Request request, int statusCode) throws Exception {
+        isValidNotificationId(request.getNotificationId());
+        String query = String.format(insertSubscription, postgresSubscription, UUID.randomUUID(), request.getSenderCode(),
+                request.getNotificationId(), statusCode, System.currentTimeMillis(), "API", statusCode,
+                System.currentTimeMillis());
+        postgreSQLClient.execute(query);
+    }
+
+    public void getSubscriptions(Request request, Response response) throws Exception {
+        List<Subscription> subscriptionList = fetchSubscriptions(request.getSenderCode());
+        response.setSubscriptions(subscriptionList);
+        response.setCount(subscriptionList.size());
+    }
+
+    public void notify(Request request, Response response, String kafkaTopic) throws Exception {
+        isValidNotificationId(request.getNotificationId());
+        eventHandler.processAndSendEvent(kafkaTopic, request);
+    }
+
+    public void getNotifications(NotificationListRequest request, Response response) throws Exception {
+        for (String key : request.getFilters().keySet()) {
+            if(!ALLOWED_NOTIFICATION_FILTER_PROPS.contains(key))
+                throw new ClientException(ErrorCodes.ERR_INVALID_NOTIFICATION_REQ, "Invalid notifications filters, allowed properties are: " + ALLOWED_NOTIFICATION_FILTER_PROPS);
+        }
+        // TODO: add filter limit condition
+        Set<Map<String, Object>> removeNotificationList = new HashSet<>();
+        notificationList.forEach(notification -> request.getFilters().keySet().forEach(key -> {
+            if (!notification.get(key).equals(request.getFilters().get(key)))
+                removeNotificationList.add(notification);
+        }));
+        notificationList.removeAll(removeNotificationList);
+        response.setNotifications(notificationList);
+        response.setCount(notificationList.size());
+    }
+
+    private void loadNotifications() throws IOException {
+        Resource resource = resourceLoader.getResource(filename);
+        ObjectMapper jsonReader = new ObjectMapper(new JsonFactory());
+        notificationList = (List<Map<String, Object>>) jsonReader.readValue(resource.getInputStream(), List.class);
+    }
+
+    private void isValidNotificationId(String notificationId) throws ClientException {
+        boolean isValid = false;
+        for (Map<String, Object> notification: notificationList) {
+            if(notificationId.equals(notification.get(TOPIC_CODE)))
+                isValid = true;
+        }
+        if(!isValid) throw new ClientException(ErrorCodes.ERR_INVALID_NOTIFICATION_TOPIC_CODE, "Invalid NotificationId." + notificationId + " is not present in the master list of notifications");
+    }
```

Comment:
Why don't you create the topic code list in init?

Reply:
updated the implementation